crates/ide/src/hover/render.rs

```diff
@@ -718,18 +718,13 @@ pub(super) fn definition(
             }
             _ => return None,
         };
-        let rendered_drop_glue = match drop_info.drop_glue {
-            DropGlue::None => "does not contain types with destructors (drop glue)",
-            DropGlue::DependOnParams => {
-                "may contain types with destructors (drop glue) depending on type parameters"
-            }
-            DropGlue::HasDropGlue => "contain types with destructors (drop glue)",
+        let rendered_drop_glue = match (drop_info.drop_glue, drop_info.has_dtor) {
+            (DropGlue::HasDropGlue, _) | (_, Some(true)) => "has Drop",
+            (DropGlue::DependOnParams, _) => "type param may have Drop",
+            _ => "no Drop",
         };
```

Comment:
"having Drop` sounds weird to me, we might as well use the term drop glue then or significant destructor

Reply:
I think I'd prefer to call it significant destructor

Reply:
I want to avoid `drop glue` (at least until https://github.com/rust-lang/reference/issues/873 is resolved), so like the idea of using destructor.

How about:

* `has destructor`
* `no destructor`
* `type param may have destructor`

What does "significant" mean in this context?

Reply:
Actually, reading through https://doc.rust-lang.org/nomicon/destructors.html, we use the phrase "needs Drop" for a type that has drop glue but no `impl Drop`.

So we could also do:

* `impl Drop`
* `needs Drop`
* `no Drop`
* `type params may need Drop`

I don't have a strong preference, but do like using `Drop` as the word that shows up in the source code.

Reply:
Slight vote/preference for `impl Drop` from me.

Reply:
`impl Drop` won't be correct though, these don't necessarily implement `Drop` when they have drop glue. But sure, needs Drop would probably work